sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md

```diff
@@ -7,6 +7,7 @@
 #### Breaking Changes
 
 #### Bugs Fixed
+* Fixed an issue when using `ChangeFeed` with `spark.cosmos.changeFeed.itemCountPerTriggerHint` causing all cosmos partitions to not be fully processed in some cases. - See [PR 42553](https://github.com/Azure/azure-sdk-for-java/pull/42553)
```

Comment:
I do not think that itemCountPerTriggerHint is relevant - in Spark streaming the EndLSN is always calcualted before the micro batch is processed. So, the functional issue that not all child partitions might have been drained until the EndLSN is still a problem even when itemCountPerTriggerHint is not specified (independent of the trigger type)

Reply:
for trigger type where we will ignore the endLSN, this issue probably will not exist, as we only depend on the hasNext checking. Probably only Triggered.Once currently

Reply:
Correct Trigger.Once might be an exception because Spark asks for ignoring the OffsetLimit. But for all other trigger types the EndLSN is calculated before the first CF requets of a micro batch is issued - independent of whethe ritemcountpertriggerhint is specified or not.

Reply:
Changed the changelog message.